### PR TITLE
Test: Adds functions for input verification, and test the demo survey

### PR DESCRIPTION
**Pull request by GabrielBruno24**
Write a test for the demo survey that will check input verification for various widgets.

Adds the following tests to the testHelpers file:
Test that input fields will not accept invalid character types Test that invalid string values will highlight the box in red Test that entering an invalid date will autocorrect it Test that clicking the continue button when not all answers are filled will not advance to the next page Test that clicking the continue button when age is under 18 will open a popup

### CODE REVIEW

File: packages/evolution-frontend/tests/ui-testing/testHelpers.ts [hunk 1]
```diff
@@ -465,7 +509,7 @@ export const inputMapFindPlaceTest: InputMapFindPlaceTest = ({ context, path, ex
         const refreshButton = context.page.locator(`id=survey-question__${newPath}_refresh`);
         await refreshButton.scrollIntoViewIfNeeded();
         await refreshButton.click();
-
+
```
Comment (tahini) [inline packages/evolution-frontend/tests/ui-testing/testHelpers.ts:512]: Remove spaces on this line. It was accidentally added.

File: packages/evolution-frontend/tests/ui-testing/testHelpers.ts [hunk 2]
```diff
@@ -628,3 +694,54 @@ export const waitTextVisible = ({ context, text, isVisible = true }: { text: Pat
...
+ *
+ * @param {Object} options - The options for the test.
+ * @param {string} options.text - The text of the button to click.
+ * @param {string} options.currentPageUrl - The URL of the current page.
+ * @param {string} options.nextPageUrl - The URL of the page that the button would normally take us too.
+ */
+export const tryToContinueWithPopup: ContinueWithInvalidEntriesTest = ({ context, text, currentPageUrl, nextPageUrl }) => {
+    test(`Clicking ${text} should open a popup and keep you on ${currentPageUrl} - ${getTestCounter(context, `${text} - ${currentPageUrl} - ${nextPageUrl}`)}`, async () => {
+        await tryToContinueOnInvalidPage({ context, text, currentPageUrl, nextPageUrl });
+        const agePopup = context.page.getByRole('dialog');
```
Comment (tahini) [inline packages/evolution-frontend/tests/ui-testing/testHelpers.ts:742]: Rename agePopup to something else, the name does not represent what this does

File: packages/evolution-frontend/tests/ui-testing/testHelpers.ts [hunk 3]
```diff
@@ -628,3 +694,54 @@ export const waitTextVisible = ({ context, text, isVisible = true }: { text: Pat
         }
     });
 };
+
+const tryToContinueOnInvalidPage: ContinueWithInvalidEntriesTest = async ({ context, text, currentPageUrl, nextPageUrl }) => {
+    const button = context.page.getByRole('button', { name: text });
+    await button.scrollIntoViewIfNeeded();
+    await button.click();
+    await expect(context.page.waitForURL(nextPageUrl, {timeout: 1000})).rejects.toThrow("Timeout")
```
Comment (tahini) [inline packages/evolution-frontend/tests/ui-testing/testHelpers.ts:702]: Is 1000ms enough to say that we did not go to the next page? Please add a FIXME above this line to validate this, or see if there's another condition. If we hit problems on slower machines, we'll know where to look at.

**Review by tahini — CHANGES_REQUESTED** [review]
Some very small things to fix, but looks good. Good idea to split the one person flow in various sections.

**Review by tahini — APPROVED** [review]

File: example/demo_survey/tests/test-one-person-helpers.ts
Comment (samuel-duhaime) [inline example/demo_survey/tests/test-one-person-helpers.ts:1]: Nice. Bonne idée de séparer par section.

**Pull request merged**